Components/FlightControl/Flash/Flash.hpp

```diff
@@ -0,0 +1,22 @@
+/**
+ * @file    Flash.hpp
+ * @brief   This file contains the class declaration for a flash device driver.
+ *
+ */
+#ifndef FLASH_BASE_HPP
+#define FLASH_BASE_HPP
+#include "SystemDefines.hpp"
+
+class Flash
+{
+public:
+    virtual void Init() = 0;
+    virtual bool Erase(uint32_t offset) = 0;
+    virtual bool Write(uint32_t offset, uint8_t* data, uint32_t len) = 0;
```

Comment:
If getting fancy, could stage writes, e.g. 
use WRITE_WORD until all`/4 == 0` 
use WRITE_HALF_WORD until all `/2 == 0` 
use WRITE_BYTE for remainder

might not be worth the overhead though 🤷